### PR TITLE
Added missing returns on error scenarios

### LINKED ISSUE #46521: typing 'continue' in a non-loop in a shader causes freezing, occasional crash
**Godot version:** 3.2.3 stable

**OS/device including version:** Windows 10

**Issue description:**

Accidentally put continue outside of a loop, Godot crashed. Attempted to reproduce I was only able to get temporary freezes.
Expected behaviour: shader should just throw an error and not render like other errors do.

**Steps to reproduce:**
write this shader:
```
shader_type spatial;

void fragment(){
	continue;
}
```

**Minimal reproduction project:**

[ShaderCrashTest.zip](https://github.com/godotengine/godot/files/6057573/ShaderCrashTest.zip)

### PR DESCRIPTION
**Pull request by rsubtil**
There were some missing returns on some error scenarios.

Fixes #46521

### CODE REVIEW

File: servers/rendering/shader_language.cpp [hunk 1]
```diff
@@ -5928,13 +5928,15 @@ Error ShaderLanguage::_parse_block(BlockNode *p_block, const FunctionInfo &p_fun
...
 				//all is good
 				_set_error("Expected ';' after discard");
+				return ERR_PARSE_ERROR;
 			}
 
 			p_block->statements.push_back(flow);
 		} else if (tk.type == TK_CF_BREAK) {
 			if (!p_can_break) {
 				//all is good
 				_set_error("Breaking is not allowed here");
```
Comment (Calinou) [inline servers/rendering/shader_language.cpp:5938]:
```suggestion
				_set_error("'break' is not allowed outside of a loop or 'switch' statement");
```

File: servers/rendering/shader_language.cpp [hunk 2]
```diff
@@ -5961,6 +5964,7 @@ Error ShaderLanguage::_parse_block(BlockNode *p_block, const FunctionInfo &p_fun
 			if (!p_can_continue) {
 				//all is good
 				_set_error("Continuing is not allowed here");
```
Comment (Calinou) [inline servers/rendering/shader_language.cpp:5966]:
```suggestion
				_set_error("'continue' is not allowed outside of a loop");
```

**Review by Calinou — COMMENTED** [review]
Maybe we can improve the error messages while we're at it :slightly_smiling_face:

File: servers/rendering/shader_language.cpp (hunk 1 above)
Comment (Ev1lbl0w) [reply servers/rendering/shader_language.cpp:5938]: Judging from the [docs](https://docs.godotengine.org/en/3.2/tutorials/shading/shading_reference/shading_language.html#flow-control), it seems those are the only scenarios where it's used, yup.

File: servers/rendering/shader_language.cpp (hunk 2 above)
Comment (Ev1lbl0w) [reply servers/rendering/shader_language.cpp:5966]: _ditto_

**Comment by Ev1lbl0w** [comment]
Fixed; I also removed the earlier comments, as they don't make much sense anymore.

**Review by Chaosus — APPROVED** [review]
Yeah, this is the correct fix, which prevents shader crashes in these cases.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.

**Pull request merged**